plugin_ide.core/src/com/github/rustdt/ide/core/engine/RustSourceModelManager.java

```diff
@@ -57,49 +52,32 @@ protected SourceFileStructure doCreateNewData() throws CommonException, Operatio
 			
 			Location fileLocation = structureInfo.getLocation();
 			
-			String describeOutput = getDescribeOutput(source, fileLocation);
-			if(describeOutput == null) {
-				return null;
+			Optional<String> describeOutput = new RustParseDescribeLauncher(toolManager, cm).getDescribeOutput(source, fileLocation);
+			if(!describeOutput.isPresent()) {
+				// Absent case only relevant if DevelopmentCodeMarkers.TESTS_MODE is set. Would like to remove this workaround.
+				return null; // null means outline is removed.
 			}
 			
 			try {
 				RustParseDescribeParser parseDescribe = new RustParseDescribeParser(fileLocation, source);
-				SourceFileStructure newStructure = parseDescribe.parse(describeOutput);
-				if(newStructure.getParserProblems().size() > 0 && newStructure.getChildren().isEmpty()) {
-					
+				SourceFileStructure newStructure = parseDescribe.parse(describeOutput.get());
+				
+				boolean keepPreviousStructure = !newStructure.getParserProblems().isEmpty() && newStructure.getChildren().isEmpty();
+				if(keepPreviousStructure) {
 					SourceFileStructure previousStructure = structureInfo.getStoredData().getOrNull();
 					if(previousStructure != null) {
-						// Use elements from previous structure:
-						Indexable<StructureElement> previousElements = 
-								StructureElement.cloneSubTree(previousStructure.getChildren());
+						Indexable<StructureElement> previousElements = previousStructure.cloneSubTree();
 						
-						newStructure = new SourceFileStructure(fileLocation, previousElements, 
-							newStructure.getParserProblems());
+						return new SourceFileStructure(previousElements, newStructure.getParserProblems());
 					}
 				}
+				System.out.println("RustSourceModelManager - File touched: " + fileLocation);
```

Comment:
My intention is to remove this line once we agree that the *Open Type* feature works. A better solution would be to use debug logging. Do we have the infrastructure for that?